### PR TITLE
Add missing preprocessing forms

### PR DESCRIPTION
**Pull request by gitjannes**
Add the missing preprocessing forms in methods/data_preprocessing.py

### CODE REVIEW

File: backend/protzilla/methods/data_preprocessing.py [hunk 1]
```diff
@@ -393,6 +445,41 @@ class NormalisationByReferenceProtein(DataPreprocessingStep):
     operation = "normalisation"
     method_description = "Normalise data by reference protein"
 
+    def create_form(self):
```
Comment (ronjakrg) [inline backend/protzilla/methods/data_preprocessing.py:481]: Sind die neuen Forms schon im Run-Screen eingebaut? (Sorry, bin noch nicht 100%ig in der neuen Forms-Logik drin)
Ich wollte den Step hier testen und das schienen die Input Parameter noch nicht integriert zu sein, wäre aber top wenn ich das auch testen könnte :)
Screenshot from 2025-05-11 12-47-57

File: backend/protzilla/methods/data_preprocessing.py [hunk 2]
```diff
@@ -393,6 +445,41 @@ class NormalisationByReferenceProtein(DataPreprocessingStep):
...
+        return Form(
+            label="Normalisation data by reference protein",
```
Comment (ronjakrg) [inline backend/protzilla/methods/data_preprocessing.py]: Hier müsste glaube ich "Normalise data by reference protein" stehen statt "Normalisation"

File: backend/protzilla/methods/data_preprocessing.py [hunk 3]
```diff
@@ -393,6 +445,41 @@ class NormalisationByReferenceProtein(DataPreprocessingStep):
...
+            input_fields=[
+                FormDivider("A function to perform protein-intensity normalisation in reference to "
+                    "a selected protein on your dataframe. Normalises the data on the level "
+                    "of each sample. Divides each intensity by the intensity of the chosen "
+                    "reference protein in each sample. Samples where this value is zero "
+                    "will be removed and returned separately.A function to perform "
```
Comment (ronjakrg) [inline backend/protzilla/methods/data_preprocessing.py]: Hier steht die Beschreibung irgendwie mehrfach drin?

File: backend/protzilla/methods/data_preprocessing.py [hunk 4]
```diff
@@ -402,6 +489,48 @@ class ImputationByMinPerDataset(DataPreprocessingStep):
     operation = "imputation"
     method_description = "Impute missing values by the minimum per dataset"
 
+    def create_form(self):
+        return Form(
+            label="Imputation by minimum per dataset",
+            input_fields=[
+                NumberField(
+                    name="shrinking_value",
+                    label="A function to impute missing values for each protein by taking into account "
```
Comment (ronjakrg) [inline backend/protzilla/methods/data_preprocessing.py]: Hier ist die Beschreibung in das Label gerutscht, es sollte vermutlich "Shrinking value" als Label sein

File: backend/protzilla/methods/data_preprocessing.py [hunk 5]
```diff
@@ -411,6 +540,48 @@ class ImputationByMinPerProtein(DataPreprocessingStep):
     operation = "imputation"
     method_description = "Impute missing values by the minimum per protein"
 
+    def create_form(self):
+        return Form(
+            label="Imputation by minimum per protein",
+            input_fields=[
+                FloatField(
+                    name="shrinking_value",
+                    label="A function to impute missing values for each protein by taking into account data from each protein. "
```
Comment (ronjakrg) [inline backend/protzilla/methods/data_preprocessing.py]: Hier das gleiche wie oben, die Beschreibung sollte nicht in das Label des Inputfelds

File: backend/protzilla/methods/data_preprocessing.py [hunk 6]
```diff
@@ -420,6 +591,45 @@ class ImputationByMinPerSample(DataPreprocessingStep):
     operation = "imputation"
     method_description = "Impute missing values by the minimum per sample"
 
+    def create_form(self):
+        return Form(
+            label="Imputation by minimum per sample",
+            input_fields=[
+                FloatField(
+                    name="shrinking_value",
+                    label="Sets missing intensity values to the smallest measured value for each sample",
```
Comment (ronjakrg) [inline backend/protzilla/methods/data_preprocessing.py]: Same here

File: backend/protzilla/methods/data_preprocessing.py [hunk 7]
```diff
@@ -432,6 +642,43 @@ class SimpleImputationPerProtein(DataPreprocessingStep):
...
 
+    def create_form(self):
+        return Form(
+            label="Imputation per Protein",
+            input_fields=[
+                DropdownField(
+                    name="strategy",
+                    label="Strategy",
+                    value=SimpleImputerStrategyType.mean,
+                    options=SimpleImputerStrategyType,
```
Comment (ronjakrg) [inline backend/protzilla/methods/data_preprocessing.py:688]: Hier fänd ich es cool, wenn die Options "Mean", "Median" und "Most frequent" geschrieben wären, also weniger code-like :)
Screenshot from 2025-05-11 12-55-26

File: backend/protzilla/methods/data_preprocessing.py [hunk 8]
```diff
@@ -432,6 +642,43 @@ class SimpleImputationPerProtein(DataPreprocessingStep):
         "sklearn.SimpleImputer class"
     )
 
+    def create_form(self):
+        return Form(
+            label="Imputation per Protein",
```
Comment (ronjakrg) [inline backend/protzilla/methods/data_preprocessing.py:682]: Das Label wird richtig angezeigt, aber in der Sidebar & im Choose Step Modal steht "SimpleImputer", was zwar korrekt bzgl. Implementierungsdetails ist, aber nicht intuitiv für den User ist.
Screenshot from 2025-05-11 12-56-46
Könnten wir den Step nicht überall eher "Imputation per protein metrics" nennen? Dann wird auch klar, dass man hier verschiedene Kennzahlen für die Imputation wählen kann

File: backend/protzilla/methods/data_preprocessing.py [hunk 9]
```diff
@@ -506,5 +804,29 @@ class FilterPeptidesByPEPThreshold(DataPreprocessingStep):
...
+                FloatField(
+                    name="treshold",
+                    label="Threshold value for PEP",
+                    value=0,
+                    min=0,
+                ),
+                DropdownField(
+                    name="peptide_df",
+                    label="peptide_df",
+                    options=EmptyEnum, #WTF happens here? Jannes, 01.05.2025
```
Comment (ronjakrg) [inline backend/protzilla/methods/data_preprocessing.py]: Das sollte noch angepasst werden, haha :) Also I guess hier kann man verschiede Protein-Dataframes auswählen, falls man mehrere importiert hat?

File: backend/protzilla/methods/data_preprocessing.py [hunk 10]
```diff
@@ -18,6 +18,8 @@
 from backend.protzilla.steps import Step, StepManager
 from backend.protzilla.form import *
 
+class EmptyEnum(Enum):
```
Comment (ronjakrg) [inline backend/protzilla/methods/data_preprocessing.py:21]: Das muss dann vermutlich raus, sieht aus wie ein Platzhalter für unten, oder?

File: backend/protzilla/methods/data_preprocessing.py [hunk 11]
```diff
@@ -18,6 +18,8 @@
 from backend.protzilla.steps import Step, StepManager
 from backend.protzilla.form import *
```
Comment (ronjakrg) [inline backend/protzilla/methods/data_preprocessing.py:20]: (Hier an random Stelle im Code angemerkt, damit es als Thread und nicht als Kommentar im PR auftaucht:)
image
So sieht die Auswahl im Add Step Modal aus und ich hätte dazu folgende Vorschläge:
1. Die "Min per ..." Methods sollten untereinander stehen
2. Das Umbenennen des Steps mit dem SimpleImputer hatte ich unten schon geschrieben
3. Das "Filter Peptides" ist irgendwie bisschen komisch am Ende, weiter oben gibt es ja schon "Filter Proteins" und "Filter Samples", denke dann passt das von der Reihenfolge besser da oben hin. Man könnte auch "Filter Proteins & Peptides" als eine gemeinsame Kategorie machen, da die beiden einzelnen Kategorien aktuell nur jeweils einen Step haben und auch tendenziell nicht viel mehr bekommen werden.

**Review by ronjakrg — CHANGES_REQUESTED** [review]
Super cool, dass Du Dir die übrigen Preprocessing Steps vorgenommen hast! :partying_face: Bis auf einen Step konnte ich alle im Run-Screen testen und habe im Review ein paar Anmerkungen und Verbesserungsvorschläge angemerkt. :)

File: backend/protzilla/methods/data_preprocessing.py (hunk 3 above)
Comment (gitjannes) [reply backend/protzilla/methods/data_preprocessing.py]: War auch im Original so hahah. Habs jetzt nur einmal drin 👍

File: backend/protzilla/methods/data_preprocessing.py (hunk 4 above)
Comment (gitjannes) [reply backend/protzilla/methods/data_preprocessing.py]: Ist im Original auch so, ich packe das jetzt aber mal in einen FormDivider

File: backend/protzilla/methods/data_preprocessing.py (hunk 5 above)
Comment (gitjannes) [reply backend/protzilla/methods/data_preprocessing.py]: Lösung wie oben :)

File: backend/protzilla/methods/data_preprocessing.py (hunk 6 above)
Comment (gitjannes) [reply backend/protzilla/methods/data_preprocessing.py]: Wie oben

File: backend/protzilla/methods/data_preprocessing.py (hunk 7 above)
Comment (gitjannes) [reply backend/protzilla/methods/data_preprocessing.py:688]: Geht glaube mit größeren Änderungen im Backend mit einher, mache ne Karte auf 👍

File: backend/protzilla/methods/data_preprocessing.py (hunk 8 above)
Comment (gitjannes) [reply backend/protzilla/methods/data_preprocessing.py:682]: Hmm, also die Sidebar Anzeige habe ich jetzt geändert, bei der anderen Sache weiß ich aber nicht, wie zielführend die ist. Die einzelnen Steps sind glaube schon mti Absicht so genannt.

File: backend/protzilla/methods/data_preprocessing.py (hunk 9 above)
Comment (gitjannes) [reply backend/protzilla/methods/data_preprocessing.py]: Yes und EmptyEnum wird halt nur als Platzhalter genutzt, weil dann da dynamisch die Options berechnet werden :) 
Jonas hattes mir dann erklärt

File: backend/protzilla/methods/data_preprocessing.py (hunk 10 above)
Comment (gitjannes) [reply backend/protzilla/methods/data_preprocessing.py:21]: Soll eig bleiben.

File: backend/protzilla/methods/data_preprocessing.py (hunk 1 above)
Comment (gitjannes) [reply backend/protzilla/methods/data_preprocessing.py:481]: Jep mein Fehler, hatte da was übersehen :)

File: backend/protzilla/methods/data_preprocessing.py (hunk 11 above)
Comment (gitjannes) [reply backend/protzilla/methods/data_preprocessing.py:20]: zu 1. : liegt glaube nicht an der bearbeiteten Datei, hier sind die schön untereinander alphabetisch sortiert.
zu 2. : wie oben
zu 3. : einzelne Kategorien finde ich schon toll, aber ja, es sollte dann trotzdem untereinander gruppiert werden.

Hab Karten für 1 & 3 im Notion erstellt.

Comment (ronjakrg) [reply backend/protzilla/methods/data_preprocessing.py:20]: Passt, können wir uns wann anders vornehmen!

File: backend/protzilla/methods/data_preprocessing.py (hunk 8 above)
Comment (ronjakrg) [reply backend/protzilla/methods/data_preprocessing.py:682]: Da hatten wir ja im Büro schon drüber gesprochen, dass "SimpleImputer" halt die Implementierung ist, aber ein User sich nichts darunter vorstellen kann. Aber das können wir auch wann anders fixen

**Review by ronjakrg — APPROVED** [review]
Änderungen sehen gut aus, danke dass du für die weiterführenden Sachen Tickets erstellt hast! :)

**Review by pchklm — APPROVED** [review]
